CondTools/Hcal/plugins/CastorDumpConditions.cc

```diff
@@ -130,45 +133,15 @@ CastorDumpConditions::analyze(const edm::Event& iEvent, const edm::EventSetup& i
    for(std::vector<std::string>::const_iterator it=mDumpRequest.begin();it!=mDumpRequest.end();it++)
       std::cout << *it << std::endl;
 
-    if (std::find (mDumpRequest.begin(), mDumpRequest.end(), std::string ("ElectronicsMap")) != mDumpRequest.end())
-      dumpIt(new CastorElectronicsMap(), new CastorElectronicsMapRcd(), iEvent,iSetup,"ElectronicsMap");
-
-    if (std::find (mDumpRequest.begin(), mDumpRequest.end(), std::string ("QIEData")) != mDumpRequest.end())
-      dumpIt(new CastorQIEData(), new CastorQIEDataRcd(), iEvent,iSetup,"QIEData"); 
-
-    if (std::find (mDumpRequest.begin(), mDumpRequest.end(), std::string ("Pedestals")) != mDumpRequest.end()) 
-      dumpIt(new CastorPedestals(), new CastorPedestalsRcd(), iEvent,iSetup,"Pedestals");
-
-    if (std::find (mDumpRequest.begin(), mDumpRequest.end(), std::string ("PedestalWidths")) != mDumpRequest.end())
-      dumpIt(new CastorPedestalWidths(), new CastorPedestalWidthsRcd(), iEvent,iSetup,"PedestalWidths");
-
-    if (std::find (mDumpRequest.begin(), mDumpRequest.end(), std::string ("Gains")) != mDumpRequest.end())
-      dumpIt(new CastorGains(), new CastorGainsRcd(), iEvent,iSetup,"Gains");
-
-    if (std::find (mDumpRequest.begin(), mDumpRequest.end(), std::string ("GainWidths")) != mDumpRequest.end())
-      dumpIt(new CastorGainWidths(), new CastorGainWidthsRcd(), iEvent,iSetup,"GainWidths");
-
-    if (std::find (mDumpRequest.begin(), mDumpRequest.end(), std::string ("ChannelQuality")) != mDumpRequest.end())
-      dumpIt(new CastorChannelQuality(), new CastorChannelQualityRcd(), iEvent,iSetup,"ChannelQuality");
-
-    if (std::find (mDumpRequest.begin(), mDumpRequest.end(), std::string ("RecoParams")) != mDumpRequest.end())
-      dumpIt(new CastorRecoParams(), new CastorRecoParamsRcd(), iEvent,iSetup,"RecoParams");
-      
-    if (std::find (mDumpRequest.begin(), mDumpRequest.end(), std::string ("SaturationCorrs")) != mDumpRequest.end())
-      dumpIt(new CastorSaturationCorrs(), new CastorSaturationCorrsRcd(), iEvent,iSetup,"SaturationCorrs");
-
-/*
-   ESHandle<CastorPedestals> p;
-   iSetup.get<CastorPedestalsRcd>().get(p);
-   CastorPedestals* mypeds = new CastorPedestals(*p.product());
-   std::ostringstream file;
-   std::string name = "CastorPedestal";
-   file << file_prefix << name.c_str() << "_Run" << iEvent.id().run()<< ".txt";
-   std::ofstream outStream(file.str().c_str() );
-   std::cout << "CastorDumpConditions: ---- Dumping " << name.c_str() << " ----" << std::endl;
-   CastorDbASCIIIO::dumpObject (outStream, (*mypeds) );
-
-*/   
+   dumpIt<CastorElectronicsMap , CastorElectronicsMapRcd> (mDumpRequest, iEvent,iSetup,"ElectronicsMap" );
```

Comment:
please add a comment that the dump request is now checked within the function (a decent code reduction, but potentially confusing for future observers)